x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/sagemaker/SageMakerService.java

```diff
@@ -128,7 +129,7 @@ public void infer(
         boolean stream,
         Map<String, Object> taskSettings,
         InputType inputType,
-        TimeValue timeout,
+        @Nullable TimeValue timeout,
```

Comment:
Hmm I thought the timeout is defaulted in the `InferenceAction` https://github.com/elastic/elasticsearch/blob/a4a271415d882fb32ee402f316c3b45afaa3d897/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java#L182

Can it be null here?

Reply:
We should be defaulting it there too I think:

```
    static TimeValue parseTimeout(RestRequest restRequest) {
        return restRequest.paramAsTime(InferenceAction.Request.TIMEOUT.getPreferredName(), InferenceAction.Request.DEFAULT_TIMEOUT);
    }
```

I think we should consider it a bug if it's null once it gets to the `infer()` calls. We should make sure it's defaulted prior to those calls.